src/mem/slicc/symbols/StateMachine.py

```diff
@@ -64,6 +64,7 @@
     "DMASequencer": "DMASequencer",
     "RubyPrefetcher": "RubyPrefetcher",
     "Cycles": "Cycles",
+    "Addr": "Addr",
```

Comment:
> Could you explain how this doesn't break things since `Addr` is already a type that SLICC understands? Or, could you explain why this change is needed?
>

As the commit suggests this is just needed to be able to use the Addr type as a python Param for the CacheController. Even if SLICC recognises Addr as a type, you can't currently use it as a parameter. See below for further info
 
> What worries me is that there are a number of places with code like `self.symtab.find("Addr", Type)`. Does specifying this in the `python_class_map` not have any effect on how SLICC understands this type?

That was my understanding. The python_class_map is only used for type checking the Param type for the CacheController

```
 270         code(
 271             """
 272 from m5.params import *
 273 from m5.SimObject import SimObject
 274 from m5.objects.Controller import RubyController
 275  
 276 class $py_ident(RubyController):
 277     type = '$py_ident'
 278     cxx_header = 'mem/ruby/protocol/${c_ident}.hh'
 279     cxx_class = 'gem5::ruby::$py_ident'
 280 """
 281         )
 282         code.indent()
 283         for param in self.config_parameters:  
 284             dflt_str = ""
 285  
 286             if param.rvalue is not None:      
 287                 dflt_str = str(param.rvalue.inline()) + ", "
 288                                                                                                                                                                                                                                                                               
 289             if param.type_ast.type.c_ident in python_class_map:
 290                 python_type = python_class_map[param.type_ast.type.c_ident]
 291                 code(
 292                     '${{param.ident}} = Param.${{python_type}}(${dflt_str}"")'
 293                 )
 294  
 295             else:
 296                 self.error(
 297                     "Unknown c++ to python class conversion for c++ "
 298                     "type: '%s'. Please update the python_class_map "
 299                     "in StateMachine.py",     
 300                     param.type_ast.type.c_ident,
 301                 )

```